### PR TITLE
TWAI: Add receive example (Copy of #7430)

### PR DESCRIPTION
**Pull request by Dazza0**
Copy of https://github.com/espressif/arduino-esp32/pull/7430 but squashed and rebased.

### CODE REVIEW

**Comment by Dazza0** [comment]
@designer2k2 I've rebased and squashed your PR. PTAL.

**Comment by VojtechBartoska** [comment]
@PilnyTomas Can you please do one more "quick" review and after that, we can merge it.

File: libraries/ESP32/examples/TWAI/TWAIreceive/TWAIreceive.ino [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  twai_general_config_t g_config = TWAI_GENERAL_CONFIG_DEFAULT((gpio_num_t)TX_PIN, (gpio_num_t)RX_PIN, TWAI_MODE_LISTEN_ONLY);
+  twai_timing_config_t t_config = TWAI_TIMING_CONFIG_500KBITS();  //Look in the api-reference for other speed sets.
+  twai_filter_config_t f_config = TWAI_FILTER_CONFIG_ACCEPT_ALL();
+
+  // Install TWAI driver
+  if (twai_driver_install(&g_config, &t_config, &f_config) == ESP_OK) {
+    Serial.println("Driver installed");
+  } else {
+    Serial.println("Failed to install driver");
+    return;
```
Comment (me-no-dev) [inline libraries/ESP32/examples/TWAI/TWAIreceive/TWAIreceive.ino:50]: if this fails, `setup()` will return and start the `loop()` which will check for alerts on non-working bus. I suggest setting a flag that TWAI started OK, which will allow then the code in `loop()` to execute (else just delay and do nothing)

File: libraries/ESP32/examples/TWAI/TWAIreceive/TWAIreceive.ino [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  }
+
+  // Start TWAI driver
+  if (twai_start() == ESP_OK) {
+    Serial.println("Driver started");
+  } else {
+    Serial.println("Failed to start driver");
+    return;
```
Comment (me-no-dev) [inline libraries/ESP32/examples/TWAI/TWAIreceive/TWAIreceive.ino:58]: same as above

File: libraries/ESP32/examples/TWAI/TWAIreceive/TWAIreceive.ino [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      Serial.printf(" %d = %02x,", i, message.data[i]);
+    }
+    Serial.println("");
+  }
+}
+
+void loop() {
+  if (!driver_installed) {
+    // Driver not installed
+    Serial.println("TWAI driver not installed!");
```
Comment (me-no-dev) [inline libraries/ESP32/examples/TWAI/TWAIreceive/TWAIreceive.ino:95]: you can replace this line with a simple `delay(1000);` which will leave the reason as last printed line and sleep from then on. As is, this line will be printed as fast as the UART can push it out in a loop

**Review by me-no-dev — APPROVED** [review]

**Pull request merged**